orc8r/protos/bootstrapper.proto

```diff
@@ -71,3 +73,88 @@ service Bootstrapper {
 }
+
+// --------------------------------------------------------------------------
+// Protos for CloudRegistration and Registration servicer
+// --------------------------------------------------------------------------
+// CloudRegistration creates tokens for operators to give to AGWs to speed up registration workflow. It also validates tokens
+service CloudRegistration {
+  // GetToken saves and sends out a token related to input:networkID and input:logicalID
+  // It will refresh the token if input:refresh is true or token is expired
+  rpc GetToken(GetTokenRequest) returns (GetTokenResponse) {}
+
+  // GetInfoForGatewayRegistration sends out information needed for gateways to register themselves
+  rpc GetGatewayRegistrationInfo(GetGatewayRegistrationInfoRequest) returns (GetGatewayRegistrationInfoResponse) {}
+
+  // GetGatewayPreregisterInfo sends out information that has been set when the operator preregistered
+  // the device with networkID and logicalID
+  rpc GetGatewayDeviceInfo(GetGatewayDeviceInfoRequest) returns (GetGatewayDeviceInfoResponse) {}
+}
+
+// Registration services a southbound AGW register endpoint that takes in the registration token made by CloudRegistration
+// and the rest of gateway info needed to register an AGW
+service Registration {
+  // Register should be called by the gateway device
+  rpc Register(RegisterRequest) returns (RegisterResponse) {}
+}
+
+message GetTokenRequest {
+  GatewayDeviceInfo gateway_device_info = 1;
+  // refresh is true if a new token should be generated regardless of old token timeout
+  bool refresh = 2;
+}
+
+message GetTokenResponse {
+  // token is a nonce prepended by bootstrapper.tokenPrepend
+  string token = 1;
+  google.protobuf.Timestamp timeout = 2;
+}
+
+message GetGatewayRegistrationInfoRequest {}
+
+message GetGatewayRegistrationInfoResponse {
+  string root_ca = 1;
+  string domain_name = 2;
+}
+
+message GetGatewayDeviceInfoRequest {
+  // token is a nonce prepended by bootstrapper.tokenPrepend
+  string token = 1;
+}
+
+message GetGatewayDeviceInfoResponse {
+  oneof response {
+    GatewayDeviceInfo gateway_device_info = 1;
+    string error = 2;
+  }
+}
+
+message RegisterRequest {
+  // token is a nonce prepended by bootstrapper.tokenPrepend
+  string token = 1;
+  AccessGatewayID hwid = 2;
+  // challenge_key is gateway's long-term public key
+  ChallengeKey challenge_key = 3;
+}
+
+message RegisterResponse {
+  oneof response {
+    // control_proxy is the OK that the gateway has registered properly
+    // EG magma/lte/gateway/configs/control_proxy.yml
+    string control_proxy = 1;
+    string error = 2;
+  }
+}
+
+// Basic info that an AGW needs to register
+message GatewayDeviceInfo {
+  string network_id = 1;
+  string logical_id = 2;
+}
+
+// Info saved that is keyed by token. This is what is saved on CloudRegistration's initial register call (GetToken)
+message TokenInfo {
+  GatewayDeviceInfo gateway_device_info = 1;
+  string nonce = 2;
```

Comment:
We want to provide doc comments for every proto field

Reply:
Yep. Makes things more approachable, and since the comments get replicated into generated code, its a double-win